core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -258,6 +258,11 @@ Optional<TableExecuteHandle> getTableHandleForExecute(
 
     Optional<TableLayout> getNewTableLayout(Session session, String catalogName, ConnectorTableMetadata tableMetadata);
 
+    /**
+     * Returns the table metadata with the effective Trino types specific for the connector
+     */
+    ConnectorTableMetadata getNewTableEffectiveMetadata(Session session, String catalogName, QualifiedObjectName tableName, List<ColumnMetadata> columns, Map<String, Object> properties, Optional<String> comment);
```

Comment:
That's meant to be used (and is used) only for CTAS and not for regular CT.
The method name should reflect that.

cc @martint for any naming suggestion

Reply:
`getConnectorTableMetadataForCTAS`
`getEffectiveConnectorTableMetadataForCTAS`
`getConnectorTableMetadataForCreateTableAsSelect`
`getCreateTableAsSelectConnectorTableMetadata`

Out of those my preference would be `getConnectorTableMetadataForCreateTableAsSelect`.

WDYT?

Reply:
What is the point of passing properties and comment?